api/routes/utils.js

```diff
@@ -485,42 +484,50 @@ var utils = {
       }
 
       amount *= -1;
+
+      // Dividends pay 10%
+      var dividendsPercentage = 0.1;
+      var newInvestorReps = portfolio[index].reps + amount;
       var investment = portfolio[index].investments[j];
-      var prevAmount = investment.amount;
 
-      // Revenue is the fraction sold times the valuation
-      var revenue = Math.floor(amount / prevAmount * investment.valuation);
-      var roi = (revenue - amount)/amount;
+      var prevInvestmentAmount = investment.amount;
+      var prevInvestmentPercentage = investment.percentage;
+      var newInvestmentAmount = prevInvestmentAmount - amount;
+
+      // newPercentage / newAmount = oldPercentage / oldAmount (Proportional)
+      var newInvestmentPercentage = Math.floor((newInvestmentAmount * investment.percentage) / prevInvestmentAmount);
 
-      // Adjust the investor's reps
-      portfolio[index].reps += revenue;
+      // New dividends equal old dividends reduced by the difference between
+      // what the investment contributed previously and what it contributes now
+      var dividendsChange = (
```

Comment:
I think the dividends will still change, but only be the fraction of the revoke. i.e. revoking a quarter of the investment should lower my dividends by a quarter.

This means that the dividend should not actually be updated here. The job will handle that itself, since the payout is based on the percentage and the amount of the investment.